### PR TITLE
Confluence App Integration in MindsDB

### PR DESCRIPTION
**Pull request by bala-ceg**
## Description
Implemented the Confluence handler as per the new application handler framework

**Fixes** #5354
## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📄 This change requires a documentation update

## What is the solution?
- [x] Confluence spaces table for a given confluence-hosted url instance
  - [x] Support LIMIT
  - [x] Support WHERE
  - [x] Support ORDER BY
  - [x] Support column selection

### Working Screenshots

Screenshot 2023-04-19 at 8 57 34 AM

Screenshot 2023-04-19 at 9 03 14 AM

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation or created issues to update them.
- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] I have shared a short loom video or screenshots demonstrating any new functionality.

### CODE REVIEW

**Review by MinuraPunchihewa — APPROVED** [review]
This looks great to me!

**Comment by bala-ceg** [comment]
Thanks for taking the time to review this PR @MinuraPunchihewa

File: mindsdb/integrations/handlers/confluence_handler/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+Confluence is a collaborative documentation tool,it can be used to host Wiki pages. In this handler,python client of api is used and more information about this python client can be found (here)[https://pypi.org/project/atlassian-python-api/]
+
+
+## Confluence Handler Initialization
+
+The Confluence handler is initialized with the following parameters:
+
+- `url`: Confluence hosted url instance
+- `confluence_api_token`: Confluence API key to use for authentication
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/confluence_handler/README.md:15]: Can we just add a link to the docs for generating the token please?

File: mindsdb/integrations/handlers/confluence_handler/confluence_table.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                    raise ValueError("Unsupported where operation for type")
+                if a_where[2] not in ["personal", "global"]:
+                    raise ValueError(
+                        f"Unsupported where argument for state {a_where[2]}"
+                    )
+                spaces_kwargs["type"] = a_where[2]
+            else:
+                raise ValueError(f"Unsupported where argument {a_where[1]}")
+
+        confluence_spaces_df = pd.json_normalize(self.handler.connect().get_all_spaces(start=0,limit=total_results)["results"])
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/confluence_handler/confluence_table.py:73]: Can we make this to be two lines since it will fails on pep8 validation?

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]

File: mindsdb/integrations/handlers/confluence_handler/confluence_table.py (hunk 2 above)
Comment (bala-ceg) [reply mindsdb/integrations/handlers/confluence_handler/confluence_table.py]: Fixed this, thanks for pointing it out, zoran

**Comment by bala-ceg** [comment]
@ZoranPandovski @MinuraPunchihewa Fixed the review comments & additionally i reduced the debug log level in `__init__.py`
Please re-review the PR

File: mindsdb/integrations/handlers/confluence_handler/README.md (hunk 1 above)
Comment (bala-ceg) [reply mindsdb/integrations/handlers/confluence_handler/README.md:14]: Done. thanks Zoran for reviewing

**Review by ZoranPandovski — APPROVED** [review]

**Pull request merged**